frontend/components/CourseActions/components/StatsModal/index.js

```diff
@@ -0,0 +1,92 @@
+import TogglerAndModal from '~/components/TogglerAndModal';
+import css from './index.scss';
+import { Link }        from 'react-router-dom';
+import api from '~/api';
+
+class StatsModal extends React.Component {
+  static propTypes = {
+    toggler: PropTypes.element.isRequired,
+    course: PropTypes.object.isRequired,
+    learners: PropTypes.array.isRequired,
+    currentUser: PropTypes.object.isRequired,
+    author: PropTypes.object.isRequired
+  }
+
+  state = {
+    learners: this.props.learners,
+    stats: [],
+    speUpdate: {}
+  }
+
+  getStats = () =>
+    api.CourseApi.getStudentsStats(
+      (spe) => spe.status === 'success',
+      { courseId: this.props.course.id, authorId: this.props.author.id }
+    )
+    .then((payload) => {
+      this.state.stats = []
+
+      for (let index = 0; index < payload.length; index++) {
+        const element = payload[index];
+        const merged = { ...this.state.learners[index], ...element };
+        this.state.stats.push(merged)
+      }
+    })
+
+  getStudentsStats = () => {
+    this.getStats();
+  }
+
+  renderTable = () =>
+    <table className="standard-table">
+      {this.getStudentsStats()}
```

Comment:
`this.getStudentStats()` should go into `componentDidMount()`, otherwise we'll be fetching the api on every UI update.

Reply:
In fact I suggest we move this method to the parent element, would be nice if the author doesn't have to wait for the stats to load after he opens the modal.

Reply:
Alright, I simplified and moved that `getStats()` business to the parent element and called it into a `componentDidMount()`.